### ISSUE TITLE
[Feature Request] Semantic highlight of each local variable with a different color

### ISSUE THREAD

**Issue by krupitskas** [microsoft/vscode-cpptools#9413]
### Feature Request

Hello! 

I really like enhanced highlighting, but I've saw feature called semantic highlighting in the CLion which is basically highlights all the local variables with a different color, which makes it easier to distinguish. 
I'm curios, is it possible to do it with intellisense? 
If it is, maybe you can guide me a bit and I would glad to contribute.
As example I attach my project, as you can see, 3 variables initialized and all of them has different color.
Thanks!
![image](https://user-images.githubusercontent.com/5683877/172254043-36b33424-7b68-4c84-b7ab-7a0c1dd27a3c.png)

**Comment by sean-mcmanus** [comment]
I don't believe this can be implemented without additional support from VS Code: https://github.com/microsoft/vscode .

@Colengms correct me me if I'm mistaken.

**Comment by Colengms** [comment]
> highlights all the local variables with a different color

Hi @krupitskas .  The C/C++ extension does return a distinct token for local variables.  You can see how tokens (both syntactic tokens from VS Code and semantic tokens from the C/C++ extension) are associated with your code using the `Developer: Inspect Editor Tokens and Scopes` command, then clicking on a keyword in your source code.


![image](https://user-images.githubusercontent.com/49173979/172261650-09bf97fb-4f49-46fc-809d-7084e24e047a.png)

If the theme you are using does not use a unique color for a token that you would like colored uniquely, you can override the colorization of that token in your local settings.  Guidance on this is  [here](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#theming)
 and (here](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide)

**Comment by sean-mcmanus** [comment]
@Colengms Yeah, I was right about to add that scope info you provided. However, they're requesting a feature from VS Code that auto-rotates the color, similar to what VS Code does with braces.

**Comment by Colengms** [comment]
Indeed, the auto-rotating of the color would require support from VS Code.  @krupitskas , consider opening a feature request against VS Code for this.

**Comment by Colengms** [comment]
We could potentially implement this in the C/C++ extension using editor 'text decorations'.  However, that's very inefficient.  If instead VS Code were to handle the coloration for us and provide an API to allow us to group tokens for the same variable, that could potentially provide much better performance.
